php/ext/google/protobuf/protobuf.c

```diff
@@ -241,30 +331,21 @@ static void php_proto_hashtable_descriptor_release(zval* value) {
   }
   efree(ptr);
 }
-#endif
-
-static PHP_RINIT_FUNCTION(protobuf) {
-  int i = 0;
-
-  ALLOC_HASHTABLE(upb_def_to_php_obj_map);
-  zend_hash_init(upb_def_to_php_obj_map, 16, NULL, HASHTABLE_VALUE_DTOR, 0);
 
-  ALLOC_HASHTABLE(ce_to_php_obj_map);
-  zend_hash_init(ce_to_php_obj_map, 16, NULL, HASHTABLE_VALUE_DTOR, 0);
+static void test_release(void* value) {
+  void* ptr = value;
+}
+#endif
 
-  ALLOC_HASHTABLE(proto_to_php_obj_map);
-  zend_hash_init(proto_to_php_obj_map, 16, NULL, HASHTABLE_VALUE_DTOR, 0);
+static initialize_persistent_descriptor_pool(TSRMLS_D) {
+  upb_inttable_init(&upb_def_to_desc_map_persistent, UPB_CTYPE_PTR);
```

Comment:
What does `upb_inttable_init` do? This seems central to the theme about creating the persistent pool.

Reply:
It initializes a custom hashtable implemented by upb.